### ISSUE TITLE
jsdoc/valid-types for "is" property

### ISSUE THREAD

**Issue by youbastard** [gajus/eslint-plugin-jsdoc#1221]
## Expected behavior
Defining a property or param named `is` should be allowed. 

## Actual behavior
When defining a param or property named `is`, the linter reports an error for the rule `jsdoc/valid-types`. 

## ESLint Config

```js
{
  "env": {
    "es2021": true
  },
  "extends": [
    "eslint:recommended"
  ],
  "plugins": [
    "jsdoc"
  ],
  "parserOptions": {
    "sourceType": "module",
    "ecmaVersion": "latest"
  },
  "rules": {
    "jsdoc/valid-types": 2
  }
}
```

## ESLint sample

```js
/**
 * @param {string} is
 * @param {string} foo
 * 
 * @returns {string}
 */
export default function (is, foo) {
    return is + foo;
}
```

I've also created a project that reproduces the issue on codesandbox.io:

https://codesandbox.io/p/devbox/festive-sun-g23dhw-y3vs47

<img width="555" alt="Screenshot 2024-04-04 at 12 47 57 PM" src="https://github.com/gajus/eslint-plugin-jsdoc/assets/544345/239f79df-31c3-4563-bb6c-91761666b8d8">

## Environment

- Node version: v20.9.0
- ESLint version: v8.57.0
- `eslint-plugin-jsdoc` version: 48.2.2

**Comment by brettz9** [comment]
The fix in #1222 doesn't handle nested use of the keyword. For that, one has to escape with quotation marks.
